src/bos/operators/utils/clients/pcs.py

```diff
@@ -118,9 +118,12 @@ def status(nodes, session=None, **kwargs):
       PowerControlException: Any non-nominal response from PCS.
       JSONDecodeError: Error decoding the PCS response
     """
+    status_bucket = defaultdict(set)
+    if not nodes:
```

Comment:
Some operators overwrite the `_run` method with their own version of it:

```
$ grep 'def _run' *.py
base.py:    def _run(self) -> None:
discovery.py:    def _run(self) -> None:
session_cleanup.py:    def _run(self) -> None:
session_completion.py:    def _run(self) -> None:
session_setup.py:    def _run(self) -> None:
status.py:    def _run(self) -> None:
```
For those operators, the `_run` method from the base class is not used (unless it is explicitly invoked as calling the base class version of the method, but I don't recall ever seeing that in the code).

For example, this call to `_run`:
https://github.com/Cray-HPE/bos/blob/64288bcfe5ee1235b5aa7edaaf27d8bd60698834/src/bos/operators/base.py#L99

It uses `self.run`, so for the operators that I listed above, it will be calling their `_run` methods, not the one from the base operator.

Reply:
In terms of your first suggestion, that is one of the things this PR does -- it causes the status operator to short circuit when it has nothing to act on. I also reviewed all of the other operators to make sure they either don't override the base operator, or that they handle this type of case (for whatever resource they act on). Are you seeing any operators left where I missed that?

But as you say, even with that, I think it makes sense to have the checks in the utility functions themselves. That way if new operators are added or those clients get used in other ways, we're still covered.